modules/extension/app-schema/app-schema/src/main/java/org/geotools/appschema/jdbc/JoiningJDBCFeatureSource.java

```diff
@@ -602,7 +615,7 @@ public String selectSQL(
                 // apply paging to the root feature if applicable
                 Collection<String> ids = new ArrayList<>();
 
-                if (isRootFeature && query.isDenormalised()) {
+                if (isRootFeature && query.isDenormalised() && !isCount) {
```

Comment:
There are several continuous IF statements with the same extra condition "&& !isCount".  You could group them in a bigger IF with only this condition for better understanding and readability on how the algorithm workflow works.

Reply:
fixed: I've put the logic applying paging with respect to the rootness of the featureType in a separate methods avoiding some code repetition and reducing the occurences of the isCount boolean